DistroLauncher/OOBE.cpp

```diff
@@ -77,152 +73,71 @@ namespace DistributionInfo
         return S_OK;
     }
 
-    // Anonimous namespace to avoid exposing internal details of the implementation.
-    namespace
+    // Saves Windows information inside Linux filesystem to supply to the OOBE.
+    // Returns empty string if we fail to generate the prefill file
+    // or the postfix to be added to the OOBE command line.
+    std::wstring PreparePrefillInfo()
     {
-
-        // Saves Windows information inside Linux filesystem to supply to the OOBE.
-        // Returns empty string if we fail to generate the prefill file
-        // or the postfix to be added to the OOBE command line.
-        std::wstring PreparePrefillInfo()
-        {
-            std::wstring commandLine;
-            const auto prefillInfo = DistributionInfo::GetPrefillInfoInYaml();
-            if (prefillInfo.empty()) {
-                return commandLine;
-            }
-
-            // Write it to a file inside \\wsl.localhost distro filesystem.
-            const std::wstring prefillFileNameDest = L"/var/tmp/prefill-system-setup.yaml";
-            const std::wstring wslPrefix = L"\\\\wsl.localhost\\" + DistributionInfo::Name;
-            std::ofstream prefillFile;
-            // Mixing slashes and backslashes that way is not a problem for Windows.
-            prefillFile.open(wslPrefix + prefillFileNameDest, std::ios::ate);
-            if (prefillFile.fail()) {
-                Helpers::PrintErrorMessage(CO_E_FAILEDTOCREATEFILE);
-                return commandLine;
-            }
-
-            prefillFile << prefillInfo;
-            prefillFile.close();
-            if (prefillFile.fail()) {
-                Helpers::PrintErrorMessage(CO_E_FAILEDTOCREATEFILE);
-                return commandLine;
-            }
-
-            commandLine += L" --prefill=" + prefillFileNameDest;
+        std::wstring commandLine;
+        const auto prefillInfo = DistributionInfo::GetPrefillInfoInYaml();
+        if (prefillInfo.empty()) {
             return commandLine;
-        } // std::wstring PreparePrefillInfo().
-
-        // OOBEStatusHandling checks the exit status of wsl-setup script
-        // and takes the required actions.
-        HRESULT OOBEStatusHandling(std::wstring_view status)
```

Comment:
I’m puzzled to not see those 2 functions now?

Reply:
Their replacements are in <https://github.com/ubuntu/WSL/blob/main/DistroLauncher/ExitStatus.cpp>.